libcaf_net/test/datagram_transport.cpp

```diff
@@ -0,0 +1,261 @@
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#define CAF_SUITE datagram_transport
+
+#include "caf/net/datagram_transport.hpp"
+
+#include "caf/net/test/host_fixture.hpp"
+#include "caf/test/dsl.hpp"
+
+#include "caf/byte.hpp"
+#include "caf/make_actor.hpp"
+#include "caf/net/actor_proxy_impl.hpp"
+#include "caf/net/endpoint_manager.hpp"
+#include "caf/net/endpoint_manager_impl.hpp"
+#include "caf/net/make_endpoint_manager.hpp"
+#include "caf/net/multiplexer.hpp"
+#include "caf/net/socket_guard.hpp"
+#include "caf/net/udp_datagram_socket.hpp"
+#include "caf/serializer_impl.hpp"
+#include "caf/span.hpp"
+
+using namespace caf;
+using namespace caf::net;
+
+namespace {
+
+constexpr string_view hello_manager = "hello manager!";
+
+class dummy_application_factory;
+
+struct fixture : test_coordinator_fixture<>, host_fixture {
+  using buffer_type = std::vector<byte>;
+
+  using buffer_ptr = std::shared_ptr<buffer_type>;
+
+  fixture() : shared_buf(std::make_shared<buffer_type>(1024)) {
+    mpx = std::make_shared<multiplexer>();
+    if (auto err = mpx->init())
+      CAF_FAIL("mpx->init failed: " << sys.render(err));
+    mpx->set_thread_id();
+    CAF_CHECK_EQUAL(mpx->num_socket_managers(), 1u);
+    if (auto err = parse("127.0.0.1:0", ep))
+      CAF_FAIL("parse returned an error: " << sys.render(err));
+    auto send_pair = unbox(make_udp_datagram_socket(ep));
+    send_socket = send_pair.first;
+    auto receive_pair = unbox(make_udp_datagram_socket(ep));
+    recv_socket = receive_pair.first;
+    ep.port(htons(receive_pair.second));
+    CAF_MESSAGE("sending message to " << CAF_ARG(ep));
+    if (auto err = nonblocking(recv_socket, true))
+      CAF_FAIL("nonblocking() returned an error: " << err);
+  }
+
+  ~fixture() {
+    close(send_socket);
+    close(recv_socket);
```

Comment:
These two sockets should go into a socket guard instead. Can be a followup, since we just extended the socket guard API in #35 to allow this.